### ISSUE TITLE
jenkins-ci-ARM-ubuntu-18-04 always fail

### ISSUE THREAD

**Issue by teawater** [kata-containers/runtime#1206]
https://github.com/kata-containers/runtime/pull/1166
I rebuilt jenkins-ci-ARM-ubuntu-18-04 a lot of times.  But always got:
```
21:43:44 go: disabling cache (/home/jenkins/.cache/go-build) due to initialization failure: open /home/jenkins/.cache/go-build/log.txt: permission denied
21:44:19 # github.com/kata-containers/runtime/cli
21:44:19 ./utils.go:78:10: undefined: unknown
21:44:19 ./utils.go:82:13: undefined: unknown
21:44:19 Makefile:497: recipe for target '/home/jenkins/workspace/kata-containers-runtime-ARM-18.04-PR/go/src/github.com/kata-containers/runtime/kata-runtime' failed
21:44:19 make: *** [/home/jenkins/workspace/kata-containers-runtime-ARM-18.04-PR/go/src/github.com/kata-containers/runtime/kata-runtime] Error 2
21:44:20 Build step 'Execute shell' marked build as failure
21:44:20 Performing Post build task...
21:44:20 Match found for :.* : True
21:44:20 Logical operation result is TRUE
21:44:20 Running script  : #!/bin/bash
21:44:20 
21:44:20 export GOPATH=$WORKSPACE/go
21:44:20 export GOROOT="/usr/local/go"
21:44:20 export PATH=${GOPATH}/bin:/usr/local/go/bin:/usr/sbin:/usr/local/bin:${PATH}
21:44:20 
21:44:20 cd $GOPATH/src/github.com/kata-containers/tests
21:44:20 .ci/teardown.sh "$WORKSPACE/artifacts"
21:44:20 
21:44:20 # And ensure the workspace tree is all owned by us before we quit, otherwise the later
21:44:20 # `delete workspace before run` may fail on file perms (as some tests leave things owned by root).
21:44:20 sudo chown -R ${USER} ${WORKSPACE}
21:44:20 sudo chgrp -R $(id -ng) ${WORKSPACE}
21:44:21 [kata-containers-runtime-ARM-18.04-PR] $ /bin/bash /tmp/jenkins1376191090923206481.sh
21:44:23 ~/workspace/kata-containers-runtime-ARM-18.04-PR/artifacts ~/workspace/kata-containers-runtime-ARM-18.04-PR/go/src/github.com/kata-containers/tests
21:44:29 ~/workspace/kata-containers-runtime-ARM-18.04-PR/go/src/github.com/kata-containers/tests
21:44:29 INFO: Checking log files
21:44:29 bash: .ci/hypervisors//filter_docker_.sh: No such file or directory
21:44:30 make -C cmd/log-parser
21:44:30 make[1]: Entering directory '/home/jenkins/workspace/kata-containers-runtime-ARM-18.04-PR/go/src/github.com/kata-containers/tests/cmd/log-parser'
21:44:30 go test .
21:44:30 go: disabling cache (/home/jenkins/.cache/go-build) due to initialization failure: open /home/jenkins/.cache/go-build/log.txt: permission denied
21:44:39 ok  	github.com/kata-containers/tests/cmd/log-parser	0.087s
21:44:39 go build -o "kata-log-parser" -ldflags "-X main.name=kata-log-parser -X main.commit="bdc1daa8cd7b0e2590f3600e3bbc5559f3ed50cd" -X main.version=0.0.1" .
21:44:39 go: disabling cache (/home/jenkins/.cache/go-build) due to initialization failure: open /home/jenkins/.cache/go-build/log.txt: permission denied
21:44:42 install -d /home/jenkins/workspace/kata-containers-runtime-ARM-18.04-PR/go/bin
21:44:42 install kata-log-parser /home/jenkins/workspace/kata-containers-runtime-ARM-18.04-PR/go/bin/kata-log-parser
21:44:42 make[1]: Leaving directory '/home/jenkins/workspace/kata-containers-runtime-ARM-18.04-PR/go/src/github.com/kata-containers/tests/cmd/log-parser'
21:44:42 time="2019-02-01T21:44:42.172481436+08:00" level=debug msg="ignoring empty file \"/home/jenkins/workspace/kata-containers-runtime-ARM-18.04-PR/go/src/github.com/kata-containers/tests/kata-ksm-throttler.log\"\n" commit=bdc1daa8cd7b0e2590f3600e3bbc5559f3ed50cd name=kata-log-parser pid=33449 source=log-parser version=0.0.1
21:44:42 time="2019-02-01T21:44:42.172877794+08:00" level=debug msg="ignoring empty file \"/home/jenkins/workspace/kata-containers-runtime-ARM-18.04-PR/go/src/github.com/kata-containers/tests/kata-proxy.log\"\n" commit=bdc1daa8cd7b0e2590f3600e3bbc5559f3ed50cd name=kata-log-parser pid=33449 source=log-parser version=0.0.1
21:44:42 time="2019-02-01T21:44:42.173086768+08:00" level=debug msg="ignoring empty file \"/home/jenkins/workspace/kata-containers-runtime-ARM-18.04-PR/go/src/github.com/kata-containers/tests/kata-runtime.log\"\n" commit=bdc1daa8cd7b0e2590f3600e3bbc5559f3ed50cd name=kata-log-parser pid=33449 source=log-parser version=0.0.1
21:44:42 time="2019-02-01T21:44:42.173302352+08:00" level=debug msg="ignoring empty file \"/home/jenkins/workspace/kata-containers-runtime-ARM-18.04-PR/go/src/github.com/kata-containers/tests/kata-shim.log\"\n" commit=bdc1daa8cd7b0e2590f3600e3bbc5559f3ed50cd name=kata-log-parser pid=33449 source=log-parser version=0.0.1
21:44:42 time="2019-02-01T21:44:42.173511176+08:00" level=debug msg="ignoring empty file \"/home/jenkins/workspace/kata-containers-runtime-ARM-18.04-PR/go/src/github.com/kata-containers/tests/kata-vc-throttler.log\"\n" commit=bdc1daa8cd7b0e2590f3600e3bbc5559f3ed50cd name=kata-log-parser pid=33449 source=log-parser version=0.0.1
21:44:42 ERROR: kata-log-parser: no log records to process
21:44:45 POST BUILD TASK : SUCCESS
21:44:45 END OF POST BUILD TASK : 0
21:44:45 Archiving artifacts
21:44:47 Setting status of 13ff7e3d51a0599d2bb5f0a4081976e17a2d10da to FAILURE with url http://jenkins.katacontainers.io/job/kata-containers-runtime-ARM-18.04-PR/42/ and message: 'Build finished. '
21:44:47 Using context: jenkins-ci-ARM-ubuntu-18-04
21:44:47 Finished: FAILURE
```

**Comment by jodh-intel** [comment]
```
21:44:19 ./utils.go:78:10: undefined: unknown
21:44:19 ./utils.go:82:13: undefined: unknown
```

That's odd since `unknown` is defined at the top of `utils.go`!?!

**Comment by grahamwhaley** [comment]
@Pennyzct - is this something you know about already?

**Comment by Pennyzct** [comment]
Hi~ @jodh-intel @grahamwhaley @teawater sorry for the delay, I and my colleagues are enjoying Chinese Spring Year holidays, so the response could be slow. ;)
`go: disabling cache (/home/jenkins/.cache/go-build) due to initialization failure: open /home/jenkins/.cache/go-build/log.txt: permission denied`, the first error should be fixed [here](https://github.com/kata-containers/ci/issues/30#issuecomment-458420536). 
the second `./utils.go:78:10: undefined: unknown` is also odd to me.

**Comment by teawater** [comment]
http://jenkins.katacontainers.io/job/kata-containers-runtime-ARM-18.04-PR/80/consoleText
/home/jenkins/workspace/kata-containers-runtime-ARM-18.04-PR/go/src/github.com/kata-containers/runtime/virtcontainers/utils
cli/kata-check_arm64.go:129:9:warning: if block ends with a return statement, so drop this else and outdent its block (move short variable declaration to its own line if necessary) (golint)
cli/kata-check_test.go:141:113:warning: undeclared name: TestDataa (unused)
cli/kata-check_test.go:141:113:warning: unused variable or constant undeclared name: TestDataa (varcheck)
virtcontainers/qemu_arm64_test.go:43:23:warning: unused variable or constant undeclared name: virt (varcheck)
virtcontainers/qemu_arm64_test.go:50:11:warning: unused variable or constant undeclared name: defaultMemSlots (varcheck)
virtcontainers/qemu_arm64_test.go:54:40:warning: unused variable or constant too few arguments in call to arm64.memoryTopology (varcheck)
virtcontainers/qemu_arm64_test.go:29:23:warning: unused variable or constant undeclared name: virt (varcheck)
cli/kata-check_test.go:141:113:warning: undeclared name: TestDataa (staticcheck)
virtcontainers/qemu_arm64_test.go:43:23:warning: unused struct field undeclared name: virt (structcheck)
virtcontainers/qemu_arm64_test.go:50:11:warning: unused struct field undeclared name: defaultMemSlots (structcheck)
virtcontainers/qemu_arm64_test.go:54:40:warning: unused struct field too few arguments in call to arm64.memoryTopology (structcheck)
virtcontainers/qemu_arm64_test.go:29:23:warning: unused struct field undeclared name: virt (structcheck)
cli/kata-check_test.go:141:113:warning: unused struct field undeclared name: TestDataa (structcheck)
Build step 'Execute shell' marked build as failure
Performing Post build task...
Match found for :.* : True
Logical operation result is TRUE
Running script  : #!/bin/bash

@Pennyzct This is what I got with the current commit.

**Comment by grahamwhaley** [comment]
Is this the same as #1218 ?
@sameo @teawater @Pennyzct - let's make sure we are all aligned on fixing the ARM CI, and not duplicating :-) thx!

**Comment by Pennyzct** [comment]
Hi~ @grahamwhaley @teawater @jodh-intel I have already tried to fix these static check errors with #1202 , when it first showed up quite a weeks ago.  ptal. ;)

**Comment by sameo** [comment]
@teawater Those static checks are now fixed with #1218 but some unit tests are now failing.
@Pennyzct travis is failing #1202, could you please take a look at #1218 instead?
